src/coprocessor/endpoint.rs

```diff
@@ -821,4 +838,34 @@ mod tests {
         }
         panic!("suppose to get ServerIsBusy error.");
     }
+
+    #[test]
+    fn test_stack_guard() {
+        let mut expr = Expr::new();
+        for _ in 0..10 {
+            let mut e = Expr::new();
+            e.mut_children().push(expr);
+            expr = e;
```

Comment:
can height 10 cause stack > 100?

Reply:
Maybe and may not. The check here is not exact the level of stack but just the nested level of messages.